commit c6e4cd418fa0bac13d19b398a5f42ca0f121531a
Author: Alexander Kurtakov <akurtako@redhat.com>
Date:   Tue Sep 26 09:17:46 2017 +0300

    Bug 524692 - Use try-with-resources in examples
    
    Change-Id: I1e23f36c6eef2f6fc087478f6a6a85735bad7a7f
    Signed-off-by: Alexander Kurtakov <akurtako@redhat.com>

4	27	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/addressbook/AddressBook.java
1	5	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/browserexample/BrowserExample.java
37	30	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/clipboard/MyTypeTransfer.java
6	16	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/BrowserTab.java
9	14	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
2	13	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/FileViewer.java
6	9	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/hoverhelp/HoverHelp.java
16	16	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
3	4	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/javaviewer/JavaViewer.java
5	10	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/paint/PaintExample.java
3	22	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/texteditor/TextEditor.java
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/addressbook/AddressBook.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/addressbook/AddressBook.java
index f82b89a..a304d1b 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/addressbook/AddressBook.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/addressbook/AddressBook.java
@@ -284,12 +284,10 @@ private void openAddressBook() {
 	Cursor waitCursor = shell.getDisplay().getSystemCursor(SWT.CURSOR_WAIT);
 	shell.setCursor(waitCursor);
 
-	FileReader fileReader = null;
-	BufferedReader bufferedReader = null;
 	String[] data = new String[0];
-	try {
-		fileReader = new FileReader(file.getAbsolutePath());
-		bufferedReader = new BufferedReader(fileReader);
+	try (FileReader fileReader = new FileReader(file.getAbsolutePath());
+			BufferedReader bufferedReader = new BufferedReader(fileReader);){
+
 		String nextLine = bufferedReader.readLine();
 		while (nextLine != null){
 			String[] newData = new String[data.length + 1];
@@ -305,17 +303,7 @@ private void openAddressBook() {
 		displayError(resAddressBook.getString("IO_error_read") + "\n" + file.getName());
 		return;
 	} finally {
-
 		shell.setCursor(null);
-
-		if(fileReader != null) {
-			try {
-				fileReader.close();
-			} catch(IOException e) {
-				displayError(resAddressBook.getString("IO_error_close") + "\n" + file.getName());
-				return;
-			}
-		}
 	}
 
 	String[][] tableInfo = new String[data.length][table.getColumnCount()];
@@ -355,9 +343,7 @@ private boolean save() {
 		lines[i] = encodeLine(itemText);
 	}
 
-	FileWriter fileWriter = null;
-	try {
-		fileWriter = new FileWriter(file.getAbsolutePath(), false);
+	try (FileWriter fileWriter = new FileWriter(file.getAbsolutePath(), false);){
 		for (String line : lines) {
 			fileWriter.write(line);
 		}
@@ -369,15 +355,6 @@ private boolean save() {
 		return false;
 	} finally {
 		shell.setCursor(null);
-
-		if(fileWriter != null) {
-			try {
-				fileWriter.close();
-			} catch(IOException e) {
-				displayError(resAddressBook.getString("IO_error_close") + "\n" + file.getName());
-				return false;
-			}
-		}
 	}
 
 	shell.setText(resAddressBook.getString("Title_bar")+file.getName());
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/browserexample/BrowserExample.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/browserexample/BrowserExample.java
index 842520b..81674ca 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/browserexample/BrowserExample.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/browserexample/BrowserExample.java
@@ -320,14 +320,10 @@ public class BrowserExample {
 				if (images == null) {
 					images = new Image[imageLocations.length];
 					for (int i = 0; i < imageLocations.length; ++i) {
-						InputStream sourceStream = clazz.getResourceAsStream(imageLocations[i]);
+						try (InputStream sourceStream = clazz.getResourceAsStream(imageLocations[i])) {
 						ImageData source = new ImageData(sourceStream);
 						ImageData mask = source.getTransparencyMask();
 						images[i] = new Image(null, source, mask);
-						try {
-							sourceStream.close();
-						} catch (IOException e) {
-							e.printStackTrace ();
 						}
 					}
 				}
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/clipboard/MyTypeTransfer.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/clipboard/MyTypeTransfer.java
index 3ecc8a8..35882cd 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/clipboard/MyTypeTransfer.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/clipboard/MyTypeTransfer.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2000, 2013 IBM Corporation and others.
+ * Copyright (c) 2000, 2017 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -10,8 +10,15 @@
  *******************************************************************************/
 package org.eclipse.swt.examples.clipboard;
 
-import java.io.*;
-import org.eclipse.swt.dnd.*;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+import org.eclipse.swt.dnd.ByteArrayTransfer;
+import org.eclipse.swt.dnd.DND;
+import org.eclipse.swt.dnd.TransferData;
 
 public class MyTypeTransfer extends ByteArrayTransfer {
 
@@ -31,18 +38,18 @@ public void javaToNative (Object object, TransferData transferData) {
 	try {
 		// write data to a byte array and then ask super to convert to pMedium
 		ByteArrayOutputStream out = new ByteArrayOutputStream();
-		DataOutputStream writeOut = new DataOutputStream(out);
-		for (MyType myType : myTypes) {
-			byte[] buffer = myType.firstName.getBytes();
-			writeOut.writeInt(buffer.length);
-			writeOut.write(buffer);
-			buffer = myType.firstName.getBytes();
-			writeOut.writeInt(buffer.length);
-			writeOut.write(buffer);
+		try (DataOutputStream writeOut = new DataOutputStream(out)) {
+			for (MyType myType : myTypes) {
+				byte[] buffer = myType.firstName.getBytes();
+				writeOut.writeInt(buffer.length);
+				writeOut.write(buffer);
+				buffer = myType.firstName.getBytes();
+				writeOut.writeInt(buffer.length);
+				writeOut.write(buffer);
+			}
+			byte[] buffer = out.toByteArray();
+			super.javaToNative(buffer, transferData);
 		}
-		byte[] buffer = out.toByteArray();
-		writeOut.close();
-		super.javaToNative(buffer, transferData);
 	} catch (IOException e) {
 	}
 }
@@ -56,23 +63,23 @@ public Object nativeToJava(TransferData transferData){
 		MyType[] myData = new MyType[0];
 		try {
 			ByteArrayInputStream in = new ByteArrayInputStream(buffer);
-			DataInputStream readIn = new DataInputStream(in);
-			while(readIn.available() > 20) {
-				MyType datum = new MyType();
-				int size = readIn.readInt();
-				byte[] name = new byte[size];
-				readIn.read(name);
-				datum.firstName = new String(name);
-				size = readIn.readInt();
-				name = new byte[size];
-				readIn.read(name);
-				datum.lastName = new String(name);
-				MyType[] newMyData = new MyType[myData.length + 1];
-				System.arraycopy(myData, 0, newMyData, 0, myData.length);
-				newMyData[myData.length] = datum;
-				myData = newMyData;
+			try (DataInputStream readIn = new DataInputStream(in)) {
+				while(readIn.available() > 20) {
+					MyType datum = new MyType();
+					int size = readIn.readInt();
+					byte[] name = new byte[size];
+					readIn.read(name);
+					datum.firstName = new String(name);
+					size = readIn.readInt();
+					name = new byte[size];
+					readIn.read(name);
+					datum.lastName = new String(name);
+					MyType[] newMyData = new MyType[myData.length + 1];
+					System.arraycopy(myData, 0, newMyData, 0, myData.length);
+					newMyData[myData.length] = datum;
+					myData = newMyData;
+				}
 			}
-			readIn.close();
 		} catch (IOException ex) {
 			return null;
 		}
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/BrowserTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/BrowserTab.java
index f30c836..48819b9 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/BrowserTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/BrowserTab.java
@@ -111,21 +111,15 @@ class BrowserTab extends Tab {
 		} else if (lastText != null) {
 			browser.setText(lastText);
 		} else {
-	        InputStream htmlStream= ControlExample.class.getResourceAsStream("browser-content.html");
-			BufferedReader br= new BufferedReader(new InputStreamReader(htmlStream));
 			StringBuffer sb= new StringBuffer(300);
-			try {
-				int read= 0;
-				while ((read= br.read()) != -1)
+
+			try (InputStream htmlStream = ControlExample.class.getResourceAsStream("browser-content.html");
+					BufferedReader br = new BufferedReader(new InputStreamReader(htmlStream))) {
+				int read = 0;
+				while ((read = br.read()) != -1)
 					sb.append((char) read);
 			} catch (IOException e) {
 				log(e.getMessage());
-			} finally {
-				try {
-					br.close();
-				} catch (IOException e) {
-					log(e.getMessage());
-				}
 			}
 			String text= sb.toString();
 			browser.setText(text);
@@ -225,15 +219,11 @@ class BrowserTab extends Tab {
 	}
 
 	public static String getContents(InputStream in) throws IOException {
-		BufferedReader br= new BufferedReader(new InputStreamReader(in));
-
 		StringBuffer sb= new StringBuffer(300);
-		try {
+		try (BufferedReader br= new BufferedReader(new InputStreamReader(in))) {
 			int read= 0;
 			while ((read= br.read()) != -1)
 				sb.append((char) read);
-		} finally {
-			br.close();
 		}
 		return sb.toString();
 	}
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
index 3e85e4c..cd093c4 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2000, 2014 IBM Corporation and others.
+ * Copyright (c) 2000, 2017 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -11,7 +11,6 @@
 package org.eclipse.swt.examples.controlexample;
 
 
-import java.io.IOException;
 import java.io.InputStream;
 import java.text.MessageFormat;
 import java.util.MissingResourceException;
@@ -191,18 +190,14 @@ public class ControlExample {
 					images = new Image[imageLocations.length];
 
 					for (int i = 0; i < imageLocations.length; ++i) {
-						InputStream sourceStream = clazz.getResourceAsStream(imageLocations[i]);
-						ImageData source = new ImageData(sourceStream);
-						if (imageTypes[i] == SWT.ICON) {
-							ImageData mask = source.getTransparencyMask();
-							images[i] = new Image(null, source, mask);
-						} else {
-							images[i] = new Image(null, source);
-						}
-						try {
-							sourceStream.close();
-						} catch (IOException e) {
-							e.printStackTrace();
+						try (InputStream sourceStream = clazz.getResourceAsStream(imageLocations[i])) {
+							ImageData source = new ImageData(sourceStream);
+							if (imageTypes[i] == SWT.ICON) {
+								ImageData mask = source.getTransparencyMask();
+								images[i] = new Image(null, source, mask);
+							} else {
+								images[i] = new Image(null, source);
+							}
 						}
 					}
 				}
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/FileViewer.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/FileViewer.java
index dcf79ea..95c8d1d 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/FileViewer.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/FileViewer.java
@@ -1358,25 +1358,14 @@ public class FileViewer {
 				//System.out.println(getResourceString("simulate.CopyFromTo.text",
 				//	new Object[] { oldFile.getPath(), newFile.getPath() }));
 			} else {
-				FileReader in = null;
-				FileWriter out = null;
-				try {
-					in = new FileReader(oldFile);
-					out = new FileWriter(newFile);
-
+				try (FileReader in = new FileReader(oldFile);
+						FileWriter out = new FileWriter(newFile);){
 					int count;
 					while ((count = in.read()) != -1) out.write(count);
 				} catch (FileNotFoundException e) {
 					return false;
 				} catch (IOException e) {
 					return false;
-				} finally {
-					try {
-						if (in != null) in.close();
-						if (out != null) out.close();
-					} catch (IOException e) {
-						return false;
-					}
 				}
 			}
 		}
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/hoverhelp/HoverHelp.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/hoverhelp/HoverHelp.java
index 63be90c..e8e2cb8 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/hoverhelp/HoverHelp.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/hoverhelp/HoverHelp.java
@@ -84,20 +84,17 @@ public class HoverHelp {
 				images = new Image[imageLocations.length];
 
 				for (int i = 0; i < imageLocations.length; ++i) {
-					InputStream stream = clazz.getResourceAsStream(imageLocations[i]);
-					ImageData source = new ImageData(stream);
-					ImageData mask = source.getTransparencyMask();
-					images[i] = new Image(display, source, mask);
-					try {
-						stream.close();
+					try (InputStream stream = clazz.getResourceAsStream(imageLocations[i])) {
+						ImageData source = new ImageData(stream);
+						ImageData mask = source.getTransparencyMask();
+						images[i] = new Image(display, source, mask);
 					} catch (IOException e) {
 						e.printStackTrace();
 					}
 				}
 			}
 		} catch (Exception ex) {
-			System.err.println(getResourceString("error.CouldNotLoadResources",
-				new Object[] { ex.getMessage() }));
+			System.err.println(getResourceString("error.CouldNotLoadResources", ex.getMessage()));
 			return null;
 		}
 
@@ -138,7 +135,7 @@ public class HoverHelp {
 	 * with the given arguments. If the key is not found,
 	 * return the key.
 	 */
-	public String getResourceString(String key, Object[] args) {
+	public String getResourceString(String key, Object... args) {
 		try {
 			return MessageFormat.format(getResourceString(key), args);
 		} catch (MissingResourceException e) {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
index 887db69..a0abea0 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
@@ -892,18 +892,18 @@ public class ImageAnalyzer {
 		ImageLoader oldLoader = loader;
 		try {
 			URL url = new URL(urlname);
-			InputStream stream = url.openStream();
-			loader = new ImageLoader();
-			if (incremental) {
-				// Prepare to handle incremental events.
-				loader.addImageLoaderListener(event -> incrementalDataLoaded(event));
-				incrementalThreadStart();
+			try (InputStream stream = url.openStream()) {
+				loader = new ImageLoader();
+				if (incremental) {
+					// Prepare to handle incremental events.
+					loader.addImageLoaderListener(event -> incrementalDataLoaded(event));
+					incrementalThreadStart();
+				}
+				// Read the new image(s) from the chosen URL.
+				long startTime = System.currentTimeMillis();
+				imageDataArray = loader.load(stream);
+				loadTime = System.currentTimeMillis() - startTime;
 			}
-			// Read the new image(s) from the chosen URL.
-			long startTime = System.currentTimeMillis();
-			imageDataArray = loader.load(stream);
-			loadTime = System.currentTimeMillis() - startTime;
-			stream.close();
 			if (imageDataArray.length > 0) {
 				currentName = urlname;
 				fileName = null;
@@ -1243,11 +1243,11 @@ public class ImageAnalyzer {
 			ImageData[] newImageData;
 			if (fileName == null) {
 				URL url = new URL(currentName);
-				InputStream stream = url.openStream();
-				long startTime = System.currentTimeMillis();
-				newImageData = loader.load(stream);
-				loadTime = System.currentTimeMillis() - startTime;
-				stream.close();
+				try (InputStream stream = url.openStream()) {
+					long startTime = System.currentTimeMillis();
+					newImageData = loader.load(stream);
+					loadTime = System.currentTimeMillis() - startTime;
+				}
 			} else {
 				long startTime = System.currentTimeMillis();
 				newImageData = loader.load(fileName);
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/javaviewer/JavaViewer.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/javaviewer/JavaViewer.java
index 6e994d1..8a51718 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/javaviewer/JavaViewer.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/javaviewer/JavaViewer.java
@@ -144,8 +144,8 @@ void open(String name) {
 
 	try {
 		FileInputStream stream= new FileInputStream(file.getPath());
-		try {
-			Reader in = new BufferedReader(new InputStreamReader(stream));
+		try (Reader in = new BufferedReader(new InputStreamReader(stream))) {
+
 			char[] readBuffer= new char[2048];
 			StringBuffer buffer= new StringBuffer((int) file.length());
 			int n;
@@ -154,8 +154,7 @@ void open(String name) {
 			}
 			textString = buffer.toString();
 			stream.close();
-		}
-		catch (IOException e) {
+		} catch (IOException e) {
 			// Err_file_io
 			String message = MessageFormat.format(resources.getString("Err_file_io"), file.getName());
 			displayError(message);
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/paint/PaintExample.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/paint/PaintExample.java
index 595af4b..5b34006 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/paint/PaintExample.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/paint/PaintExample.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2000, 2016 IBM Corporation and others.
+ * Copyright (c) 2000, 2017 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -13,7 +13,6 @@ package org.eclipse.swt.examples.paint;
 
 import static org.eclipse.swt.events.SelectionListener.widgetSelectedAdapter;
 
-import java.io.IOException;
 import java.io.InputStream;
 import java.text.MessageFormat;
 import java.util.MissingResourceException;
@@ -476,14 +475,10 @@ public class PaintExample {
 				for (Tool tool2 : tools) {
 					Tool tool = tool2;
 					String id = tool.group + '.' + tool.name;
-					InputStream sourceStream = clazz.getResourceAsStream(getResourceString(id + ".image"));
-					ImageData source = new ImageData(sourceStream);
-					ImageData mask = source.getTransparencyMask();
-					tool.image = new Image(null, source, mask);
-					try {
-						sourceStream.close();
-					} catch (IOException e) {
-						e.printStackTrace();
+					try (InputStream sourceStream = clazz.getResourceAsStream(getResourceString(id + ".image"))) {
+						ImageData source = new ImageData(sourceStream);
+						ImageData mask = source.getTransparencyMask();
+						tool.image = new Image(null, source, mask);
 					}
 				}
 				return;
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/texteditor/TextEditor.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/texteditor/TextEditor.java
index f9d18aa..35e21ef 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/texteditor/TextEditor.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/texteditor/TextEditor.java
@@ -244,18 +244,10 @@ public class TextEditor {
 		    String name = dialog.open();
 		    if (name == null)  return;
 		    fileName = name;
-		    FileInputStream file = null;
-		    try {
-		    	file = new FileInputStream(name);
+		    try (FileInputStream file = new FileInputStream(name);){
 		    	styledText.setText(openFile(file));
 		    } catch (IOException e) {
 		    	showError(getResourceString("Error"), e.getMessage()); //$NON-NLS-1$
-		    } finally {
-		    	try {
-		    		if (file != null) file.close();
-		    	} catch (IOException e) {
-		    		showError(getResourceString("Error"), e.getMessage()); //$NON-NLS-1$
-		    	}
 		    }
 		}));
 
@@ -1130,12 +1122,10 @@ public class TextEditor {
 
 	Image loadImage(Display display, String fileName) {
 		Image image = null;
-		try {
-			InputStream sourceStream = getClass().getResourceAsStream(fileName + ".ico");  //$NON-NLS-1$
+		try (InputStream sourceStream = getClass().getResourceAsStream(fileName + ".ico")) { //$NON-NLS-1$
 			ImageData source = new ImageData(sourceStream);
 			ImageData mask = source.getTransparencyMask();
 			image = new Image(display, source, mask);
-			sourceStream.close();
 		} catch (IOException e) {
 			showError(getResourceString("Error"), e.getMessage()); //$NON-NLS-1$
 		}
@@ -1245,19 +1235,10 @@ public class TextEditor {
 
 	void saveFile() {
 		if (fileName != null) {
-			FileWriter file = null;
-			try {
-				file = new FileWriter(fileName);
+			try (FileWriter file = new FileWriter(fileName);) {
 		       	file.write(styledText.getText());
-		       	file.close();
 			} catch (IOException e) {
 	        	showError(getResourceString("Error"), e.getMessage());
-	        } finally {
-	        	try {
-	        		if (file != null) file.close();
-	        	} catch (IOException e) {
-	        		showError(getResourceString("Error"), e.getMessage());
-	        	}
 	        }
 		}
 	}
